packages/components/src/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ReactElement, cloneElement, useState } from "react";
+import ReactDatePicker from "react-datepicker";
+import { XOR } from "ts-xor";
+import { Button } from "../Button";
+import { strFormatDate } from "../FormatDate";
+
+/**
+ * Disabling no-internal-modules here because we need
+ * to reach into the package to get the css file.
+ */
+// eslint-disable-next-line import/no-internal-modules
+import "react-datepicker/dist/react-datepicker.css";
+
+interface DatePickerReturnedDates {
+  readonly formatted: string;
+  readonly raw: Date;
+}
+
+interface DatePickerBaseProps {
+  /**
+   * Change handler. Will return a date string in the prefered format
+   */
+  onChange(val: DatePickerReturnedDates): void;
+}
+
+interface DatePickerModalProps extends DatePickerBaseProps {
+  /**
+   * Display the Datepicker inline instead of in a modal style
+   */
+  readonly activator?: ReactElement;
+}
+
+interface DatePickerInlineProps extends DatePickerBaseProps {
+  /**
+   * Display the Datepicker inline instead of in a modal style
+   */
+  readonly inline: boolean;
+}
+
+type DatePickerProps = XOR<DatePickerInlineProps, DatePickerModalProps>;
```

Comment:
Is there an advantage to extending and creating a new inline interface with this XOR here vs just having `inline` be optional in the base props?  Just curious :)

Reply:
In the proposal it says...

>  The reason is that we do not want the consumer to use inline and activator props at the same time.

Reply:
Ah okay, cool!